src/Strategy/Python/Poetry/PyProject.hs

```diff
@@ -0,0 +1,311 @@
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+module Strategy.Python.Poetry.PyProject (
+  PyProject (..),
+  PyProjectPoetry (..),
+  PyProjectBuildSystem (..),
+  PoetryDependency (..),
+  pyProjectCodec,
+  pyProjectBuildSystemCodec,
+  usesPoetryBuildSystem,
+  PyProjectPoetryGitDependency (..),
+  PyProjectPoetryPathDependency (..),
+  PyProjectPoetryUrlDependency (..),
+  PyProjectPoetryDetailedVersionDependency (..),
+  parseConstraintExpr,
+  getDependencies,
+) where
+
+import Data.Foldable (asum)
+import Data.Map (Map)
+import Data.Map.Strict qualified as M
+import Data.String.Conversion (toString)
+import Data.Text (Text)
+import Data.Text qualified as T
+import Text.Megaparsec
+import Text.Megaparsec.Char
+
+import Control.Monad.Combinators.Expr (Operator (..), makeExprParser)
+import Data.Functor (void)
+import Data.Maybe (fromMaybe)
+import Data.Void
+import DepTypes
+import Text.Megaparsec.Char.Lexer qualified as L
+import Toml (TomlCodec, (.=))
+import Toml qualified
+
+newtype PackageName = PackageName {unPackageName :: Text} deriving (Eq, Ord, Show)
+
+data PyProject = PyProject
+  { pyprojectBuildSystem :: Maybe PyProjectBuildSystem
+  , pyprojectPoetry :: Maybe PyProjectPoetry
+  }
+  deriving (Show, Eq, Ord)
```

Comment:
This can happen if pyproject does not use poetry, or if it has no dependencies. Lots of project use custom python modules, or configure setuptools in pyproject file for thier build backend. Here is an example: https://github.com/numpy/numpy/blob/main/pyproject.toml

Reply:
I might add this as a comment.